include/cocaine/rpc/slot/generic.hpp

```diff
@@ -0,0 +1,59 @@
+/*
+    Copyright (c) 2011-2014 Andrey Sibiryov <me@kobology.ru>
+    Copyright (c) 2011-2014 Other contributors as noted in the AUTHORS file.
+
+    This file is part of Cocaine.
+
+    Cocaine is free software; you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    Cocaine is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef COCAINE_IO_GENERIC_SLOT_HPP
+#define COCAINE_IO_GENERIC_SLOT_HPP
+
+#include "cocaine/rpc/slot.hpp"
+
+#include "cocaine/tuple.hpp"
+
+#include <asio/system_error.hpp>
+
+namespace cocaine { namespace io {
+
+template<class Event, class F>
+struct generic_slot:
```

Comment:
I'd like to see any cocaine-core example of it, so it won't look like "we just added another useless template magic".